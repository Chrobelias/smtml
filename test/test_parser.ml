open Encoding
module Batch = Solver.Batch (Z3_mappings)
module Interpret = Interpret.Make (Batch)

let () =
  (* FIXME: when y = 2.0 this fails in Z3! *)
  let script =
    {|
    (declare-fun x real)
    (declare-fun y real)
    (assert (real.eq y (real.mul x x)))
    (assert (real.eq y 4.0))
    (check-sat)
    (get-model)
    |}
  in
  Parse.from_string script |> Interpret.start |> ignore
