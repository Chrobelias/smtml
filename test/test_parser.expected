sat
(model
  (y real 4.)
  (x real -2.))